setup.py

```diff
@@ -16,6 +16,59 @@
 from setuptools import setup
 
 
+version = '2.0.0rc3'
+
+
+def get_version(default=version, template='{tag}.{distance}+{commit}{dirty}'):
+    """
+    Return a version collected from git if possible or fall back to an hard-coded
+    version otherwise.
+    """
+    try:
+        tag, distance, commit, dirty = get_git_version()
+        if not distance and not dirty:
+            # we are from a clean Git tag: use tag
+            return tag
+
+        distance = 'dev{}'.format(distance)
+        if dirty:
+            time_stamp = get_time_stamp()
+            dirty = '.dirty.' + get_time_stamp()
+        else:
+            dirty = ''
+
+        return template.format(**locals())
+    except:
+        # no git data: use default version
+        return default
+
+
+def get_time_stamp():
+    """
+    Return a numeric UTC time stamp without microseconds.
+    """
+    from datetime import datetime
+    return (datetime.isoformat(datetime.utcnow()).split('.')[0]
+            .replace('T', '').replace(':', '').replace('-', ''))
+
+
+def get_git_version():
+    """
+    Return version parts from Git or raise an exception.
+    """
+    from subprocess import check_output, STDOUT
+    # this may fail with exceptions
+    cmd = 'git', 'describe', '--tags', '--long', '--dirty',
+    version = check_output(cmd, stderr=STDOUT).strip()
+    dirty = version.endswith('-dirty')
```

Comment:
No, I was no meaning to say you should continue using `setuptools_scm`. It will take me more time (and be convincing) to shape it into what I'd expect ;-)